agent/engine/docker_image_manager.go

```diff
@@ -335,18 +350,31 @@ func (imageManager *dockerImageManager) deleteImage(imageID string, imageState *
 		seelog.Errorf("Image ID to be deleted is null")
 		return
 	}
-	seelog.Infof("Removing Image: %s", imageID)
-	err := imageManager.client.RemoveImage(imageID, removeImageTimeout)
-	if err != nil {
-		if err.Error() == imageNotFoundForDeletionError {
-			seelog.Errorf("Image already removed from the instance: %v", err)
-		} else {
-			seelog.Errorf("Error removing Image %v - %v", imageID, err)
-			delete(imageManager.imageStatesConsideredForDeletion, imageState.Image.ImageID)
-			return
+
+	whitelisted := false
+	for _, m := range imageManager.persistentImages {
+		if imageID == m {
+			seelog.Infof("Image was whitelisted for deletion, image: %s", imageID)
+			whitelisted = true
+			break
+		}
+	}
+
+	if !whitelisted {
```

Comment:
Might be useful to reverse the conditional here and just return early.  We know that we'll never want to call `imageState.RemoveImageName` for images like this anyway.

```go
if whitelisted {
	return
}
```

That lets us remove an indentation level too.